test/e2e/apimachinery/field_validation.go

```diff
@@ -0,0 +1,725 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	// ensure libs have a chance to initialize
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/onsi/ginkgo/v2"
+	_ "github.com/stretchr/testify/assert"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/test/integration/fixtures"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("FieldValidation", func() {
+	f := framework.NewDefaultFramework("field-validation")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	var client clientset.Interface
+	var ns string
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.AfterEach(func(ctx context.Context) {
+		_ = client.AppsV1().Deployments(ns).Delete(ctx, "deployment", metav1.DeleteOptions{})
+		_ = client.AppsV1().Deployments(ns).Delete(ctx, "deployment-shared-unset", metav1.DeleteOptions{})
+		_ = client.AppsV1().Deployments(ns).Delete(ctx, "deployment-shared-map-item-removal", metav1.DeleteOptions{})
+		_ = client.CoreV1().Pods(ns).Delete(ctx, "test-pod", metav1.DeleteOptions{})
+	})
+
+	/*
+		Release: v1.27
+		Testname: Server side field validation, typed object
+		Description: It should reject the request if a typed object has unknown or duplicate fields.
+	*/
+	framework.ConformanceIt("should detect unknown and duplicate fields of a typed object", func(ctx context.Context) {
+		ginkgo.By("apply creating a deployment")
+		invalidMetaDeployment := `{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "my-dep",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"unknownField": "foo",
+			"replicas": 2,
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`
+		_, err := client.CoreV1().RESTClient().Post().
+			AbsPath("/apis/apps/v1").
```

Comment:
want to make sure this won't trigger more instances of the same issue reported in https://github.com/kubernetes/kubernetes/issues/115276, where running e2e tests with a non `/` server address doesn't work with some tests

I *think* `AbsPath` appends to the server base, but it would be good to verify that

Reply:
If I'm understanding that bug correctly, the issue has to do with the `RequestURI` method which is stripping out parts of the path and then adding back the uri blindly.

And you're asking to confirm that `AbsPath()` appends to the request's RESTClient's base directly? (Which is what it seems like it does here, unless I'm misunderstanding the ask: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/request.go#L324)

Reply:
yeah, thanks for checking